pkg/deploy/dev-workspace/sync.go

```diff
@@ -0,0 +1,258 @@
+		// Also, our Sync functionality requires the scheme to have the type we want to persist registered.
+		// In case of cert-manager objects, we don't want that because we would have to depend
+		// on cert manager, which would require us to also update operator-sdk version because cert-manager
+		// uses extension/v1 objects. So, we have to go the unstructured way here...
+		return readAndSyncUnstructured(deployContext, DevWorkspaceCertificateFile)
+	}
+	return true, nil
+}
+
+func syncDwConfigMap(deployContext *deploy.DeployContext) (bool, error) {
+	configMap := &corev1.ConfigMap{}
+	err := readK8SObject(DevWorkspaceConfigMapFile, configMap)
+	if err != nil {
+		return false, err
+	}
+	return syncObject(deployContext, configMap, DevWorkspaceNamespace)
+}
+
+func syncDwDeployment(deployContext *deploy.DeployContext) (bool, error) {
+	deployment := &appsv1.Deployment{}
+	err := readK8SObject(DevWorkspaceDeploymentFile, deployment)
+	if err != nil {
+		return false, err
+	}
+
+	devworkspaceControllerImage := util.GetValue(deployContext.CheCluster.Spec.DevWorkspace.ControllerImage, deploy.DefaultDevworkspaceControllerImage(deployContext.CheCluster))
+	devWorkspaceController := deployment.Spec.Template.Spec.Containers[0]
+	devWorkspaceController.Image = devworkspaceControllerImage
+	for _, env := range devWorkspaceController.Env {
+		if env.Name == "RELATED_IMAGE_devworkspace_webhook_server" {
+			env.Value = devworkspaceControllerImage
+			break
+		}
+	}
+
+	return syncObject(deployContext, deployment, DevWorkspaceNamespace)
+}
+
+func readAndSyncObject(deployContext *deploy.DeployContext, yamlFile string, obj metav1.Object, namespace string) (bool, error) {
+	err := readK8SObject(yamlFile, obj)
+	if err != nil {
+		return false, err
+	}
+
+	return syncObject(deployContext, obj, namespace)
+}
+
+func readAndSyncUnstructured(deployContext *deploy.DeployContext, yamlFile string) (bool, error) {
+	obj := &unstructured.Unstructured{}
+	err := readK8SUnstructured(yamlFile, obj)
+	if err != nil {
+		return false, err
+	}
+
+	return createUnstructured(deployContext, obj)
+}
+
+func createUnstructured(deployContext *deploy.DeployContext, obj *unstructured.Unstructured) (bool, error) {
+	check := &unstructured.Unstructured{}
+	check.SetGroupVersionKind(obj.GroupVersionKind())
+
+	err := deployContext.ClusterAPI.Client.Get(context.TODO(), client.ObjectKey{Name: obj.GetName(), Namespace: obj.GetNamespace()}, obj)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			check = nil
+		} else {
+			return false, err
+		}
+	}
+
+	if check == nil {
+		err = deployContext.ClusterAPI.Client.Create(context.TODO(), obj)
+		if err != nil {
+			if apierrors.IsAlreadyExists(err) {
+				return false, nil
+			}
+			return false, err
+		}
+	}
+
+	return true, nil
+}
+
+func syncObject(deployContext *deploy.DeployContext, obj2sync metav1.Object, namespace string) (bool, error) {
+	obj2sync.SetNamespace(namespace)
+
+	runtimeObject, ok := obj2sync.(runtime.Object)
```

Comment:
Might be worth switching to `client.Object` if it's in this version of controller-runtime, as this'll have to happen eventually on a controller-runtime dependency version bump.

I believe `client.Object` combines `runtime.Object` and `metav1.Object` so it might eliminate the type-cast below.

Reply:
I assume it depends on https://github.com/eclipse-che/che-operator/pull/1064
Anyway I would love to move it out of the current PR scope.
I'll keep it in mind and address next time.